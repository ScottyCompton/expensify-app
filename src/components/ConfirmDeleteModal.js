import React from 'react';
import Modal from 'react-modal';

const ConfirmDeleteModal = (props) => (
    <Modal
        isOpen={props.isOpen || false}
        contentLabel="Confirm Delete"
        onRequestClose={props.handleCloseModal}
        closeTimeoutMS={200}
        className="modal"
    >
        <p className="modal__title">Delete This Expense?</p>
        <button className="button btnCancel button--withspace button--secondary" onClick={props.handleCloseModal}>Cancel</button>
        <button className="button btnConfirm button--withspace button--warning" onClick={props.handleRemove}>Confirm</button>
    </Modal>
);

export default ConfirmDeleteModal;