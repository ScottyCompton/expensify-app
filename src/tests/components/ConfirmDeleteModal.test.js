import React from 'react';
import { shallow } from 'enzyme';
//import testData from '../fixtures/expenses';
import ConfirmDeleteModal from '../../components/ConfirmDeleteModal';


let wrapper, isOpen, handleCloseModal, handleRemove;

beforeEach(() => {
    isOpen = true;
    handleCloseModal = jest.fn();
    handleRemove = jest.fn();

    wrapper = shallow(<ConfirmDeleteModal isOpen={isOpen} handleCloseModal={handleCloseModal} handleRemove={handleRemove} />);
});


test('Should render delete expense modal', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should render empty modal div',() => {
    wrapper.setProps({isOpen: false});
    expect(wrapper).toMatchSnapshot();
});


test('Should close the modal window', () => {
    wrapper.find('.btnCancel').simulate('click');
    expect(handleCloseModal).toHaveBeenCalled();
});


test('Should close the modal window after selecting remove', () => {
    wrapper.find('.btnConfirm').simulate('click');
    expect(handleRemove).toHaveBeenCalled();
});

