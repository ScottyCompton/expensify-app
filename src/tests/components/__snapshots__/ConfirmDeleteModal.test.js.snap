// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render delete expense modal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm Delete"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <p
    className="modal__title"
  >
    Delete This Expense?
  </p>
  <button
    className="button btnCancel button--secondary"
    onClick={[Function]}
  >
    Cancel
  </button>
  <button
    className="button btnConfirm button--warning"
    onClick={[Function]}
  >
    Confirm
  </button>
</Modal>
`;

exports[`Should render empty modal div 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm Delete"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <p
    className="modal__title"
  >
    Delete This Expense?
  </p>
  <button
    className="button btnCancel button--secondary"
    onClick={[Function]}
  >
    Cancel
  </button>
  <button
    className="button btnConfirm button--warning"
    onClick={[Function]}
  >
    Confirm
  </button>
</Modal>
`;
